articles/session-management-in-nodejs-using-expressjs-and-express-session/index.md

```diff
@@ -0,0 +1,261 @@
+
+To learn more about their differences, check this [Session vs Cookie](https://www.javatpoint.com/session-vs-cookies) tutorial.
+
+### Prerequisites
+- Have [Node.js runtime](https://nodejs.org/en/download/) installed on your computer.
+
+- Basic knowledge on [how to use Node.js](https://nodejs.dev/learn/).
+
+- Basic understanding of how to create an [HTTP server using the Expres.js](/engineering-education/express/) library.
+
+### Setting up the required environments and libraries
+This is a Node.js project. It uses NPM to manage its libraries and dependencies. You need to create a new project directory and initialize the Node.js with `npm init –y`. This will generate a `package.json` file that will manage the dependencies for this project's tutorial.
+
+The following libraries will help us set up a Node.js session.
+
+- [Express](/engineering-education/express/) - a web framework for Node.js used to create HTTP web servers. Express provides an easy-to-use API to interact with the webserver.
+
+- [Nodemon](https://www.npmjs.com/package/nodemon) - a server utility framework for monitoring changes of the code on a text editor. It automatically restarts the server whenever code changes are detected.
+
+- [Express-session](https://www.npmjs.com/package/express-session) - an HTTP server-side framework used to create and manage a session middleware. This tutorial is all about sessions. Thus Express-session library will be the main focus.
+
+- [Cookie-parser](https://www.npmjs.com/package/cookie-parser) - used to parse cookie header to store data on the browser whenever a session is established on the server-side.
+
+- [Body-parser](https://www.npmjs.com/package/body-parser) - used to parse the body of an HTTP POST request.
+
+Install the above libraries using the comman:
+
+`npm install express express-session cookie-parser body-parser nodemon`
+
+### Express-session Options and how to use it
+To set up the session, you need to set a couple of [Express-session middleware](https://www.npmjs.com/package/express-session#sessionoptions) options, as shown below.
+
+```js
+const oneDay = 1000 * 60 * 60 * 24;
+app.use(sessions({
+    secret: "thisismysecrctekeyfhrgfgrfrty84fwir767",
+    saveUninitialized:true,
+    cookie: { maxAge: oneDay },
+    resave: false 
+}));
+```
+
+- `secret` - a random unique string key used to authenticate a session. It is stored in an environment variable and can't be exposed to the public. The key usually is long in a production environment, and it is a randomly generated string.
+
+- `resave` - takes a Boolean value. When true, it sets the session to save event without recording any changes on the sent request. `saveUninitialized` goes hand in hand with `resave`. They tell the session middleware how to react to different events in the browser.
```

Comment:
The explanation here is hard to follow. I'm not 100% sure what you are trying to convey.